### ISSUE TITLE
BaseURL Path Handling Bug

### ISSUE THREAD

**Issue by tri-adam** [sylabs/scs-key-client#28]
When a `Client` is created with a path component (ex. `https://example.com/path`), the path component ends up being ignored when building up request URLs.

The problem originates in `(*Client).newRequest`, which calls `(*url.URL).ResolveReference` to construct a request URI to `BaseURL`/`path`. When the `path` parameter begins with a `/`, it is treated as an absolute path and thus any path component of the `BaseURL` is ignored.

**Comment by tri-adam** [comment]
Upon investigation, there are three methods which need to be considered. The URI path component of two of these are constrained by the [OpenPGP HTTP Keyserver Protocol (HKP)](https://tools.ietf.org/html/draft-shaw-openpgp-hkp-00). To make this bug more specific:

- `(*Client).PKSAdd()` - the absolute path is set to `/pks/add` ([ref](https://tools.ietf.org/html/draft-shaw-openpgp-hkp-00#section-4)). This is consistent with the current behaviour.
- `(*Client).PKSLookup()` - the absolute path is built up of the base request `/pks/lookup`, followed by any variables ([ref](https://tools.ietf.org/html/draft-shaw-openpgp-hkp-00#section-3)). This is consistent with the current behaviour.
- `(*Client).GetVersion()` - should be constructed by appending `/version` to any path component specified by `(*Config).BaseURL`. This is **not** consistent with the current behaviour.

I also took a brief look at how the `gpg` command-line behaves when interacting with an HKP key server whose URI includes path component. This command:

```
gpg --keyserver pool.sks-keyservers.net/test --search-keys blah
```

Results in an HTTP `GET` request to `/pks/lookup?op=index&options=mr&search=blah`.

TLDR; this bug includes the behaviour of `(*Client).GetVersion()` only. The current behaviour of `(*Client).PKSAdd()` and `(*Client).PKSLookup()` is correct and should not be modified.
